script/Deploy.s.sol

```diff
@@ -0,0 +1,164 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.28;
+
+import "forge-std/Script.sol";
+import "forge-std/console.sol";
+import "openzeppelin-foundry-upgrades/Upgrades.sol";
+
+import "../contracts/EthStorageContractM1.sol";
+import "../contracts/EthStorageContractM1L2.sol";
+import "../contracts/EthStorageContractM2.sol";
+import "../contracts/EthStorageContractM2L2.sol";
+
+contract Deploy is Script {
+    uint256 private deployerPrivateKey;
+    address private deployer;
+    string private contractName;
+
+    function setUp() public {
+        deployerPrivateKey = vm.envUint("PRIVATE_KEY");
+        deployer = vm.addr(deployerPrivateKey);
+        console.log("Deployer address:", deployer);
+
+        contractName = vm.envString("CONTRACT_NAME");
+        console.log("Contract name:", contractName);
+
+        _validateContractName(contractName);
+        _validateEnvironmentVariables(contractName);
+    }
+
+    function run() external {
+        address owner = vm.envOr("OWNER_ADDRESS", deployer);
+        uint256 startTime = block.timestamp;
+
+        string memory contractFQN;
+        bytes memory constructorData;
+        bytes memory initData;
+
+        (contractFQN, constructorData, initData) = _getDeploymentData(contractName, deployer, startTime);
+        Options memory opts;
+        opts.constructorData = constructorData;
+
+        vm.startBroadcast(deployerPrivateKey);
+        address proxy = Upgrades.deployTransparentProxy(contractFQN, owner, initData, opts);
+        vm.stopBroadcast();
+
+        console.log("Proxy address:", proxy);
+        address proxyAdmin = Upgrades.getAdminAddress(proxy);
+        console.log("Proxy admin address:", proxyAdmin);
+        address impl = Upgrades.getImplementationAddress(proxy);
+        console.log("Implementation address:", impl);
+    }
+
+    function upgrade() external {
+        address proxyAddress = vm.envAddress("PROXY");
+        console.log("Proxy address:", proxyAddress);
+        uint256 startTime = vm.envUint("START_TIME");
+
+        string memory contractFQN;
+        bytes memory constructorData;
+        (contractFQN, constructorData,) = _getDeploymentData(contractName, deployer, startTime);
+        Options memory opts;
+        opts.constructorData = constructorData;
+
+        opts.referenceBuildInfoDir = vm.envString("REFERENCE_BUILD_INFO_DIR");
+        opts.referenceContract = vm.envString("REFERENCE_CONTRACT");
+        vm.startBroadcast(deployerPrivateKey);
+        Upgrades.upgradeProxy(proxyAddress, contractFQN, "", opts);
```

Comment:
If the ownerAddress is a Gnosis Safe, we need to follow the steps outlined [here](https://github.com/OpenZeppelin/openzeppelin-upgrades/issues/946#issuecomment-1861378609).

I suggest we open a new PR to address this properly: for testing, we can upgrade directly; for mainnet, the owner must be a Gnosis Safe, so we should use the multisig flow.